apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreTransactionImpl.cs

```diff
@@ -123,6 +123,79 @@ public override AsyncLazyDatastoreQuery RunQueryLazilyAsync(GqlQuery gqlQuery, C
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions
+
+            };
+            var runAggregationQueryResponse = _client.RunAggregationQuery(request, callSettings);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(GqlQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions,
+
+            };
+            var runAggregationQueryResponse = _client.RunAggregationQuery(request, callSettings);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(AggregationQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions
+
+            };
+            var runAggregationQueryResponse = await _client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(GqlQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions,
+
+            };
+            var runAggregationQueryResponse = await _client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        internal AggregationQueryResults ProcessAggregationQueryResponse(RunAggregationQueryResponse runAggregationQueryResponse)
+        {
+            var results = new AggregationQueryResults(runAggregationQueryResponse.Batch.MoreResults, runAggregationQueryResponse.Batch.ReadTime);
+            foreach (var aggregationResults in runAggregationQueryResponse.Batch.AggregationResults)
+            {
+                foreach (var pair in aggregationResults.AggregateProperties)
```

Comment:
This loop can possibly be removed and if I understand correctly all of this can be done in just one loop. (I may be wrong though)